bridges/FurAffinityBridge.php

```diff
@@ -0,0 +1,914 @@
+				. $this->getInput('username-folder')
+				. '/folder/'
+				. $this->getInput('folder-id');
+		default: return parent::getURI();
+		}
+	}
+
+	public function collectData() {
+		switch($this->queriedContext) {
+		case 'Search':
+			$data = array(
+				'q' => $this->getInput('q'),
+				'perpage' => 72,
+				'rating-general' => ($this->getInput('rating-general') === true ? 'on' : 0),
+				'rating-mature' => ($this->getInput('rating-mature') === true ? 'on' : 0),
+				'rating-adult' => ($this->getInput('rating-adult') === true ? 'on' : 0),
+				'range' => $this->getInput('range'),
+				'type-art' => ($this->getInput('type-art') === true ? 'on' : 0),
+				'type-flash' => ($this->getInput('type-flash') === true ? 'on' : 0),
+				'type-photo' => ($this->getInput('type-photo') === true ? 'on' : 0),
+				'type-music' => ($this->getInput('type-music') === true ? 'on' : 0),
+				'type-story' => ($this->getInput('type-story') === true ? 'on' : 0),
+				'type-poetry' => ($this->getInput('type-poetry') === true ? 'on' : 0),
+				'mode' => $this->getInput('mode')
+			);
+			$html = $this->postFASimpleHTMLDOM($data);
+			$limit = (is_int($this->getInput('limit')) ? $this->getInput('limit') : 10);
+			$this->itemsFromSubmissionList($html, $limit);
+			break;
+		case 'Browse':
+			$data = array(
+				'cat' => $this->getInput('cat'),
+				'atype' => $this->getInput('atype'),
+				'species' => $this->getInput('species'),
+				'gender' => $this->getInput('gender'),
+				'perpage' => 72,
+				'rating_general' => ($this->getInput('rating_general') === true ? 'on' : 0),
+				'rating_mature' => ($this->getInput('rating_mature') === true ? 'on' : 0),
+				'rating_adult' => ($this->getInput('rating_adult') === true ? 'on' : 0)
+			);
+			$html = $this->postFASimpleHTMLDOM($data);
+			$limit = (is_int($this->getInput('limit-browse')) ? $this->getInput('limit-browse') : 10);
+			$this->itemsFromSubmissionList($html, $limit);
+			break;
+		case 'Journals':
+			$html = $this->getFASimpleHTMLDOM($this->getURI());
+			$limit = (is_int($this->getInput('limit')) ? $this->getInput('limit') : -1);
+			$this->itemsFromJournalList($html, $limit);
+			break;
+		case 'Single Journal':
+			$html = $this->getFASimpleHTMLDOM($this->getURI());
+			$this->itemsFromJournal($html);
+			break;
+		case 'Gallery':
+		case 'Scraps':
+		case 'Favorites':
+		case 'Gallery Folder':
+			$html = $this->getFASimpleHTMLDOM($this->getURI());
+			$limit = (is_int($this->getInput('limit')) ? $this->getInput('limit') : 10);
+			$this->itemsFromSubmissionList($html, $limit);
+			break;
+		}
+	}
+
+	private function postFASimpleHTMLDOM($data) {
+			$opts = array(
+				CURLOPT_CUSTOMREQUEST => 'POST',
+				CURLOPT_POSTFIELDS => http_build_query($data)
+			);
+			$header = array(
+				'Host: ' . parse_url(self::URI, PHP_URL_HOST),
+				'Content-Type: application/x-www-form-urlencoded',
+				'Cookie: ' . self::FA_AUTH_COOKIE
+			);
+			return getSimpleHTMLDOM($this->getURI(), $header, $opts);
```

Comment:
Sadly it returns a 200 whether the cookie is valid or not, with an error message on some pages. That could be parsed but for most lists of posts the results are just filtered and otherwise look normal with no error message, so I'm not sure how a faulty cookie could be detected properly.

I suppose sending an additional request at the beginning of `collectData` to something like the users settings page would reliably return a page that could be parsed for error messages so maybe that's an option. Do you think it'd be worth it?